test/lit/passes/local-subtyping.wast

```diff
@@ -223,4 +223,30 @@
       )
     )
   )
+
+  ;; CHECK:      (func $uses-default (param $i i32)
+  ;; CHECK-NEXT:  (local $x anyref)
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (local.get $i)
+  ;; CHECK-NEXT:   (local.set $x
+  ;; CHECK-NEXT:    (ref.func $uses-default)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (local.get $x)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $uses-default (param $i i32)
+    (local $x anyref)
+    (if
+      (local.get $i)
+      ;; The only set to this local uses a more specific type than anyref.
+      (local.set $x (ref.func $uses-default))
+    )
+    (drop
+      ;; This get may use the default value, and so we should not alter the
+      ;; type of the local.
+      (local.get $x)
```

Comment:
We could still make the type more specific and leave it nullable, couldn't we? I would think that would be worth it.

Reply:
We could, yeah. It would require some fuzzer changes - as mentioned in the comment, to ignore type differences between nulls. I slightly worry that making that change might make us miss other bugs. But I'm not sure either way.

Reply:
I pushed a change to allow default values where possible, making us only avoid making a non-nullable local where the default value is used, to show what that would look like.